File: Java/01/tasks-domain/src/main/java/org/tasks/Repositories/TasksRepository.java
package org.tasks.Repositories;

import java.util.ArrayList;
import java.util.List;
import java.util.UUID;
import java.util.concurrent.ConcurrentHashMap;
import javax.inject.Named;
import org.cqrs.Repository;
import org.commons.Services.TaskDao;

@Named("tasksRepository")
public class TasksRepository implements Repository<TaskDao> {

    private static final ConcurrentHashMap<UUID, TaskDao> _storage = new ConcurrentHashMap<UUID, TaskDao>();

    @Override
    public void Delete(UUID id) {
        if (_storage.containsKey(id)) {
            _storage.remove(id);
        }
    }

    @Override
    public List<TaskDao> GetAll() {
        return new ArrayList<>(_storage.values());
    }

    @Override
    public TaskDao GetById(UUID id) {
        if (_storage.containsKey(id)) {
            return _storage.get(id);
        }
        return null;
    }

    @Override
    public TaskDao Save(TaskDao item) {
        _storage.put(item.getId(), item);
        return item;
    }
}

File: Net/02/Tasks.Domain/Repositories/TasksRepository.cs
using Cqrs;
using System;
using System.Collections.Concurrent;
using System.Collections.Generic;
using System.Linq;

namespace Tasks.Repositories
{
    public class TasksRepository : IRepository<TaskDao, Guid>
    {
        private static ConcurrentDictionary<Guid, TaskDao> _storage = new ConcurrentDictionary<Guid, TaskDao>();

        public void Delete(Guid id)
        {
            TaskDao deleted;
            if (_storage.ContainsKey(id)) _storage.TryRemove(id, out deleted);
        }

        public List<TaskDao> GetAll()
        {
            return _storage.Values.ToList();
        }

        public TaskDao GetById(Guid id)
        {
            if (_storage.ContainsKey(id)) return _storage[id];
            return null;
        }

        public TaskDao Save(TaskDao item)
        {
            _storage[item.Id] = item;
            return item;
        }
    }
}
